From ea60034e801d35815645f64b783a6173e579418a Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Thu, 19 May 2016 16:24:40 +0800
Subject: [PATCH 0499/1722] drm/amd/dal: assign link on stream creation to
 status

Previously it was done in core_link_enable_stream, however
enable stream function is not guaranteed to be called on new
stream arrival (check pipe_need_reprogram function)

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c   | 4 ----
 drivers/gpu/drm/amd/dal/dc/core/dc_stream.c | 2 ++
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index 86c4f37c..aa96c4b 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -1659,8 +1659,6 @@ void core_link_enable_stream(struct pipe_ctx *pipe_ctx)
 
 	core_dc->hwss.enable_stream(pipe_ctx);
 
-	pipe_ctx->stream->status.link = &pipe_ctx->stream->sink->link->public;
-
 	if (pipe_ctx->stream->signal == SIGNAL_TYPE_DISPLAY_PORT_MST)
 		allocate_mst_payload(pipe_ctx);
 }
@@ -1675,7 +1673,5 @@ void core_link_disable_stream(struct pipe_ctx *pipe_ctx)
 	core_dc->hwss.disable_stream(pipe_ctx);
 
 	disable_link(pipe_ctx->stream->sink->link, pipe_ctx->stream->signal);
-
-	pipe_ctx->stream->status.link = NULL;
 }
 
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_stream.c b/drivers/gpu/drm/amd/dal/dc/core/dc_stream.c
index 44bde4c..2386bb9 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_stream.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_stream.c
@@ -81,6 +81,8 @@ static bool construct(struct core_stream *stream,
 	/* EDID CAP translation for HDMI 2.0 */
 	stream->public.timing.flags.LTE_340MCSC_SCRAMBLE = dc_sink_data->edid_caps.lte_340mcsc_scramble;
 
+	stream->status.link = &stream->sink->link->public;
+
 	return true;
 }
 
-- 
2.7.4

